test/grnstate-tests.js

```diff
@@ -0,0 +1,14 @@
+// var assert = require("chai").assert;
+// import { grnState } from "./grnstate";
+// import { updateApp } from "./update-app";
+// import { setupHandlers } from "./setup-handlers";
+
+/* TODO: Framework for front-end test
+
+var testResetNormalization = (input, propertyExpected) => {
+    var sheet = xlsx.parse(input);
+    var network = spreadsheetController.parseSheet(sheet);
+
+    assert.equal(propertyExpected, grnState.propertyValue);
+}
```

Comment:
Yes, this skeleton is in the right direction. Some notes:
- I don't think you'll need `updateApp` nor `setupHandlers`. This is strictly a unit test suite for `grnState`.
- You don't need to parse any sheets. Go straight to building a network _in memory_. We aren't testing the reading of sheets here (we already have lots of tests for that)—we are testing to make sure that `grnState` updates itself correctly.

Thus, your test code will look something like this:
```javascript
describe("The grnState model object", () => {
    it("updates the reset normalization maximum correctly", () => {
        grnState.network = {
            // Build a network here. Because reset normalization max only relies on positiveWeights and
            // negativeWeights, your test network may contain just those properties.
        };

        grnState.resetNormalizationMax.should.equal(/* whatever it should be */);
    });

    // Provide additional tests as needed.
});
```

Reply:
I see. So @dondi, does this mean I don't have to go around creating tests such as the noErrors() function in tests.js (which was my original goal with resetNormalization function), and rather should should just stick to the describe, it, etc. framework?

Reply:
Correct, `noErrors` is the exception, not the rule. This becomes apparent when you consider the answer to the question: what exactly is `noErrors` testing? The answer that question shows you that the `noErrors` design was meant to address a very specific testing need, and that need does not apply here.